app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,56 @@
+class AlbumsController < ApplicationController
+
+  def index
+    @albums = Album.all.sort_by { |album| -album.rank }
+  end
+
+  def upvote
+    album = Album.find(params[:id])
+    album.rank += 1
+    album.save
```

Comment:
:+1: There's also a cool `.increment!` method that does the same thing as lines 9-10.